src/providers/analytics/analytics.ts

```diff
@@ -1,44 +1,134 @@
 import { Injectable } from '@angular/core';
 import { AppConfigProvider } from '../app-config/app-config';
-import { IAnalyticsProvider } from './analytics.model';
+import { IAnalyticsProvider, AnalyticsEventCategories } from './analytics.model';
 import { GoogleAnalytics } from '@ionic-native/google-analytics';
 import { Platform } from 'ionic-angular';
+import { Device } from '@ionic-native/device';
+import * as moment from 'moment';
+import cryptoJs from 'crypto-js';
 
 @Injectable()
 export class AnalyticsProvider implements IAnalyticsProvider {
+  private analyticsStartupError: string = 'Error starting Google Analytics';
   googleAnalyticsKey: string;
-
+  uniqueDeviceId: string;
```

Comment:
Don't think this field is needed, seems to be only used in the function scope where it's assigned.

Reply:
it's actually used on most pages... so referenced. But - could probably be incorporated into the setCurrentPage logic and removed from the pages, would still be required though.